plugin/adsk/plugin/geomNode.cpp

```diff
@@ -0,0 +1,236 @@
+//
+// Copyright 2022 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "geomNode.h"
+
+#include <mayaUsd/fileio/jobs/meshDataReadJob.h>
+#include <mayaUsd/fileio/jobs/readJob.h>
+#include <mayaUsd/fileio/utils/meshReadUtils.h>
+#include <mayaUsd/utils/util.h>
+
+#include <pxr/usd/usdGeom/mesh.h>
+
+#include <maya/MArrayDataBuilder.h>
+#include <maya/MArrayDataHandle.h>
+#include <maya/MFnCompoundAttribute.h>
+#include <maya/MFnStringData.h>
+#include <maya/MFnTypedAttribute.h>
+#include <maya/MObject.h>
+
+namespace MAYAUSD_NS_DEF {
+// ========================================================
+
+const MTypeId MAYAUSD_GEOMNODE_ID(0x58000099);
+
+const MTypeId MayaUsdGeomNode::typeId(MayaUsd::MAYAUSD_GEOMNODE_ID);
+const MString MayaUsdGeomNode::typeName("mayaUsdGeomNode");
+
+// Attributes
+MObject MayaUsdGeomNode::filePathAttr;
+MObject MayaUsdGeomNode::rootPrimAttr;
+
+// Output attributes
+MObject MayaUsdGeomNode::outGeomAttr;
+MObject MayaUsdGeomNode::outGeomMatrixAttr;
+
+namespace {
+class OutputArrayHandler
```

Comment:
A comment explaining the purpose of the class would be nice. I get it that it's a handler for output array, but what does that do and why is it needed.